Disable hooks if instrumentation is detected

Fiouz opened the pull request:
Detect instrumentation at application binding.
New static flag in XposedBridge which acts as a "kill switch" for
user installed mods - that flag is enabled when instrumentation is
detected. My understanding is that once the initial hook (the one detecting instrumentation) is fired, the VM was already forked so altering a static variable doesn't affect other past or future processes.

If you are instrumentating an application, you're likely to already have the source code, hence won't be needing to alter the behavior using Xposed.
Also, current Xposed implementation doesn't handle the way instrumentation loads classes from another classloader.

http://forum.xda-developers.com/showpost.php?p=41688269&postcount=1517

Fiouz committed:
Disable hooks if instrumentation is detected

Detect instrumentation at application binding.
New static flag in XposedBridge which acts as a "kill switch" for
user installed mods - that flag is enabled when instrumentation is
detected.

rovo89 commented:
Thank you very much! The detection is exactly how I planned it. Using a static flag also seems fine. However, I will probably move the evaluation of this flag to handleHookedMethod() and make it behave as if no callbacks were found. This is the central place where all hooks come across, so it should really kill all of the callbacks. What do you think?

Fiouz committed:
Disable hooks earlier at handleHookedMethod

Instead of just preventing user mods from initializing, disable all
hooks ability by always delegating to the original method.
This is done as soon as the Xposed native part calls the Java part
in XposedBridge.handleHookedMethod()

Fiouz commented:
Good point! I modified the code to reflect that.

rovo89 commented:
I changed a few more things (especially catching the InvocationTargetException and throwing the cause is important) and pushed it as one commit: b411f285000f900f5b8fba9fa09ece5b7ab21a1d
Thanks again!
